src/Hook_Main_Loop.c

```diff
@@ -41,6 +42,13 @@ MainLoop_AfterRender(void *message_list)
             ResponseTimeFrame = Frame + ResponseTimeInterval;
             Send_Response_Time();
         }
-    }
 
+        if (DisableChat)
+        {
+            for (int i = 0; i < countof(SessionClass_ChatEnabled); i++)
+            {
+                SessionClass_ChatEnabled[i] = false;
+            }
+        }
```

Comment:
Is it necessary to re-assign these each frame? If not, a place other than the main loop would be more suitable.

Reply:
This is a "quick and dirty" solution to prevent user from re-enabling the chat from alliances menu. A TODO could be added for future reference about a better implementation, i.e. hooking the function that handles the chat toggle.